core/src/main/scala/contextual/contextual.scala

```diff
@@ -91,7 +91,7 @@ trait Interpolator extends Interpolator.Parts { interpolator =>
   /** The macro evaluator that defines what code will be generated for this `Interpolator`. The
     * default implementation constructs a new runtime `Contextual` object, and invokes the
     * `evaluate` method on the `Interpolator`. */
-  def evaluator(contexts: Seq[Ctx], contextual: Contextual[StaticPart]): contextual.context.Tree = {
+  def evaluator(contexts: Seq[Ctx], contextual: Contextual[StaticPart]): contextual.context.Expr[Any] = {
```

Comment:
I think we should leave the return type as `context.Tree`. You'll need to implement a method, say, `compat.macroReturn`, which always returns `context.Tree`, but takes `context.Expr` in the Scala 2.10 implementation, and `context.Tree` in the 2.11+ version.